src/core/utils/Renderer.ts

```diff
@@ -15,6 +22,14 @@ export class Renderer {
     // list of char nodes to aggregate:
     _currentCharNodes: VNode[] = [];
     _lastCreatedNode: DOMElement | Node;
+    _nextEndOffset = 0;
+    _nextStartOffset = 0;
+    _rangeToSet: RangeToSet;
+    _waitedForFormat = false;
+    _delayingRange = {
+        start: false,
+        end: false,
```

Comment:
In the renderer implementation I suggested on the other commit, I think the range nodes, when encountered, should just be stored in a variable and treated separately when the rest of the nodes have been rendered. I'm not reviewing the rendering part of this commit since I expect much will change because of the review of earlier commits (and also because that PR is huge ! :p)